lib/ts/recipe/thirdparty/providers/utils.ts

```diff
@@ -1,30 +1,13 @@
-import { verify, VerifyOptions } from "jsonwebtoken";
-import jwksClient from "jwks-rsa";
+import * as jose from "jose";
 
 export async function verifyIdTokenFromJWKSEndpoint(
     idToken: string,
     jwksUri: string,
-    otherOptions: VerifyOptions
+    otherOptions: jose.JWTVerifyOptions
 ): Promise<any> {
-    const client = jwksClient({
-        jwksUri,
-    });
-    function getKey(header: any, callback: any) {
-        client.getSigningKey(header.kid, function (_, key: any) {
-            var signingKey = key.publicKey || key.rsaPublicKey;
-            callback(null, signingKey);
-        });
-    }
+    const jwks = jose.createRemoteJWKSet(new URL(jwksUri));
```

Comment:
just to confirm - since this function will be called many times (each time a social auth login happens with apple), this won't end up creating multiple instances caches for their jwks right?

Reply:
no, it creates a single instance... but then immediately discards it. which is not technically a problem (and the way it used to work), but I'll fix it to re-use the cache anyway.